### ISSUE TITLE
[Docs] Doubt regarding the complexity for computing the predictive distribution

### ISSUE THREAD

**Issue by Akella17** [cornellius-gp/gpytorch#1109]
Firstly, I would like to thank the researchers behind this the GPyTorch library as I personally found it very helpful in one of my papers. I was going through the [GPyTorch paper](https://papers.nips.cc/paper/7985-gpytorch-blackbox-matrix-matrix-gaussian-process-inference-with-gpu-acceleration.pdf) and came across the following doubts that need some clarification:
1) Am I right in assuming that the ***modified batched CG algorithm*** is only useful when we want to do kernel learning, i.e. optimizing the prior kernel's hyperparameters over its covariance family?
2) If there is no kernel learning, can I assume that the *negative log marginal likelihood loss* or its derivative is not computed?
3) Would the complexity of computing the predictive distribution alone be *O(p\*M)*, where *p* is the number of CG iterations and *M* is the time to multiply with *K_xx*?
4) In my implementation, the prior covariance consists of the sum of a linear kernel and an RBF kernel (with SKI framework, KISS_GP). So does the RBF kernel use Kronecker or Toeplitz algebra during KISS-GP (in GPyTorch's implementation of KISS-GP)? Could you also mention the time and space complexities for computing *(K_{rbf-SKI} + K_{linear})^{-1} v*, where *v* is a vector?

**Comment by gpleiss** [comment]
1) Depends on what you mean by "only useful" :) It's designed to compute all the terms for the MLL and its derivative in a single call. If you are just concerned with computing a predictive distribution, then all you care about computing are matrix solves - and so regular (batched) CG suffices. Internally in GPyTorch we still use the same CG implementation for both the MLL and for the predictive distribution.

2) "can I assume that the negative log marginal likelihood loss or its derivative is not computed?" The MLL and its derivative are computed when you call `loss = -mll(output, targets); loss.backward()`.

3) "Would the complexity of computing the predictive distribution alone be O(p*M)"? Yes

4) "So does the RBF kernel use Kronecker or Toeplitz algebra during KISS-GP"? Yes.
The complexity for `(K_{rbf-SKI} + K_{linear})^{-1} v` is `O(pM)`, where `M` is the amount of time to do a matrix multiplication with `(K_{rbf-SKI} + K_{linear})`. The linear kernel takes `O(N)` for a matrix multiplication, and `K_{rbf-SKI}` takes `O(N + m log m)`.

**Comment by Akella17** [comment]
Thanks for clarifying my questions. 
1) Am I right in assuming that the complexity of the following computation is *O(p\*M)* (I am not computing the MLL loss)?
```Python
model.train()
likelihood.train()
# result = (K_{rbf-SKI} + K_{linear} + \sigma^2 I)^{-1} v
result = likelihood(model(training_inputs)).lazy_covariance_matrix.inv_matmul(v)
```
where ```M = O(N + m log m)```. Also, the storage complexity is ```O(N)``` for the linear kernel and ```O(N+m)``` for the RBF kernel, right? ```N: Num. of training inputs, m: Num. of inducing points```

**Comment by gpleiss** [comment]
Yes - both of those are correct.
